### PR TITLE
Refactor/extend testing

### LINKED ISSUE #15: Test Sort Functionality
Test all buttons in SortBox to ensure planets change properly
Be sure to utilize test data/intercepts!

### LINKED ISSUE #16: Test Detailed View of Planets
When a user clicks a planet:
- [ ] Name of Planet
- [ ] All pertinent planet details are shown/tested for.
- [ ] A return button is present/functional
- [ ] Images have alt text

### PR DESCRIPTION
**Pull request by ConnorAndersonLarson**
# PR PlanetParty
**Connor A-L, Alex T & Katie B**

### What does this do?

- [ ] Fix
- [ ] Feature
- [X] Refactor
- [ ] Style
- [X] Testing

### Are there any known bugs?

- [X] No
- [ ] Yes (if so please disclose in Notes for Reviewer & Next Iterations)

### Summary of the changes
Adds tests for various pages and features of the app as well as moving through pages. Does not test for unique data on all items but does check to make sure items are there. Divided into three files for different points of interest.

### What ticket(s) do the changes resolve?
Close #15 
Close #16 
Please Link Issues to this PR as necessary.

### Notes for the Reviewer
Tests should all be passing. Testing is using a fixture of test data that is very similar to actual data.

### How to Test Changes?
When switching to this branch in one tab run `npm start` for the local host and in another tab run `npm run cypress` to open cypress and test different portions of the site.

### Next Iterations
Test api calls and error handling once those updates are added to main

### CODE REVIEW

**Review by alexthompson207 — APPROVED** [review]
Awesome work this the testing! It seems like you a very thorough job testing!

**Comment by alexthompson207** [comment]
I think next up in terms of testing is testing sad paths for API calls!

File: cypress/integration/single-planet.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    cy.visit(`${baseURL}venus`)
+      .get('.planet-info-list').should('contain', 'Distance from the sun:')
+      .get('.planet-info-list').should('contain', 'Diameter:')
+      .get('.planet-info-list').should('contain', 'Length of day:')
+      .get('.planet-info-list').should('contain', 'Number of moons:')
+    cy.visit(`${baseURL}uranus`)
+      .get('.planet-info-list').should('contain', 'Mass:')
+      .get('.planet-info-list').should('contain', 'Gravity:')
+      .get('.planet-info-list').should('contain', 'Length of year:')
+  })
```
Comment (knees4bees) [inline cypress/integration/single-planet.js:37]: I think this is the spot where you suggested adding some specific numbers to make sure that the right data is coming through and in the right format. I think that's a good idea!

File: cypress/integration/single-planet.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+describe('Single Planet Testing', () => {
+  const baseURL = 'http://localhost:3000/'
+
+  beforeEach(() => {
+    cy.fixture('testPlanets.json')
+      .then(planets => {
+        cy.intercept('https://dry-plains-91502.herokuapp.com/planets', {
```
Comment (knees4bees) [inline cypress/integration/single-planet.js]: This is intercepting the old API so we'll need to replace it with the URL for the new one.

File: cypress/integration/sort-page.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      .get('.planet-card').eq(7).should('contain', 'Jupiter')
+    cy.get('label').eq(4).click()
+      .get('.planet-card').eq(0).should('contain', 'Jupiter')
+      .get('.planet-card').eq(1).should('contain', 'Saturn')
+      .get('.planet-card').eq(4).should('contain', 'Earth')
+      .get('.planet-card').eq(5).should('contain', 'Mars')
+      .get('.planet-card').eq(7).should('contain', 'Venus')
+  })
+  it('Should show data after sorting', () => {
+    cy.get('.planet-card').eq(0).should('not.contain', '57909227 km')
```
Comment (knees4bees) [inline cypress/integration/sort-page.js:34]: Oh, nice way to test that the sort order changed!

File: cypress/integration/single-planet.js (hunk 2 above)
Comment (alexthompson207) [reply cypress/integration/single-planet.js]: Good catch on this!

**Review by knees4bees — APPROVED** [review]
Very thorough, Connor! Nice.

File: cypress/integration/single-planet.js (hunk 2 above)
Comment (ConnorAndersonLarson) [reply cypress/integration/single-planet.js]: good catch, fixed!

**Pull request merged**